src/realm-settings/ProfilesTab.tsx

```diff
@@ -1,12 +1,184 @@
-import React from "react";
-import { PageSection } from "@patternfly/react-core";
-
+import React, { useState } from "react";
+import {
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  Label,
+  PageSection,
+  ToolbarItem,
+} from "@patternfly/react-core";
+import { Divider, Flex, FlexItem, Radio, Title } from "@patternfly/react-core";
+import { CodeEditor, Language } from "@patternfly/react-code-editor";
+import { KeycloakDataTable } from "../components/table-toolbar/KeycloakDataTable";
+import { ListEmptyState } from "../components/list-empty-state/ListEmptyState";
+import { useTranslation } from "react-i18next";
+import { useAdminClient } from "../context/auth/AdminClient";
+import { useConfirmDialog } from "../components/confirm-dialog/ConfirmDialog";
+import { useRealm } from "../context/realm-context/RealmContext";
+import { cellWidth, IFormatterValueType } from "@patternfly/react-table";
+import { useAlerts } from "../components/alert/Alerts";
+import { Link } from "react-router-dom";
+import type ClientProfilesRepresentation from "@keycloak/keycloak-admin-client/lib/defs/clientProfilesRepresentation";
 import "./RealmSettingsSection.css";
 
 export const ProfilesTab = () => {
+  const { t } = useTranslation("realm-settings");
+  const adminClient = useAdminClient();
+  const { realm: realmName } = useRealm();
+  const { addAlert, addError } = useAlerts();
+  const [profiles, setProfiles] = useState<ClientProfilesRepresentation>();
+  const [show, setShow] = useState(false);
+
+  const loader = async () => {
+    const clientProfiles = await adminClient.clientPolicies.listProfiles({
+      realm: realmName,
+      includeGlobalProfiles: true,
+    });
+    setProfiles(clientProfiles);
+    const allClientProfiles = clientProfiles.globalProfiles?.concat(
+      clientProfiles.profiles!
+    );
+
+    return allClientProfiles ?? [];
+  };
+
+  function formatChange(show: boolean) {
+    setShow(show);
+  }
+
+  function createNewProfile() {
+    // create a new profile
+  }
+
+  const [toggleDeleteDialog, DeleteConfirm] = useConfirmDialog({
+    titleKey: t("deleteClientProfileConfirmTitle"),
+    messageKey: t("deleteClientProfileConfirm"),
+    continueButtonLabel: "delete",
+    continueButtonVariant: ButtonVariant.danger,
+    onConfirm: async () => {
+      try {
+        // delete client profile here
+        addAlert(t("deleteClientSuccess"), AlertVariant.success);
+      } catch (error) {
+        addError(t("deleteClientError"), error);
+      }
+    },
+  });
+
+  function cellFormatter(name?: IFormatterValueType) {
+    return (
+      <>
+        <Link to={""} key={`link-${name}`}>
+          {name}
+        </Link>{" "}
+        {profiles?.globalProfiles?.length! > 0 && (
```

Comment:
Using the non-null assertion operator here can cause issues at run-time. Make sure to check the value first.
```suggestion
        {profiles?.globalProfiles && profiles.globalProfiles.length > 0 && (
```